Java/Rahul/InsertionSort.java

```diff
@@ -0,0 +1,26 @@
+/* Online Java Compiler and Editor */
+public class InsertionSort{
+
+     public static void main(String []args){
+         int[] input = new int[]{10,3,8,9,5};
+          insertionSort(input);
+         for (int i = 0; i < input.length; i++)
+            System.out.println("Element at index " + i
+                               + " : " + input[i]);
+
+     }
+     
+     public static int[] insertionSort(int[] input){
+         
+        for (int j=1 ;j<input.length;j++){
+            for(int i = j ; i > 0 ;i--){
+                if(input[i] < input[i-1]){
+                    int temp = input[i-1];
+                    input[i-1] = input[i];
+                    input[i] = temp;
```

Comment:
your inner loop continues to compare values even after insertion is done. Can you think about ways to stop that and make your code little bit more faster ?

Reply:
Will try

Reply:
That is expected right?
In insertion we need to insertions until it gets to first place

Reply:
It does not have to go to the first place all the time. If `current value` is greater the `i-1`, it should stop iterating and insert it into that position. This make our code little bit faster by skipping few iterations.

Reply:
Got it
Will change
Thank you